jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -192,11 +195,24 @@ public void close() {
     @Override
     public io.opentracing.Tracer.SpanBuilder addReference(
         String referenceType, io.opentracing.SpanContext referencedContext) {
-      if (referencedContext instanceof SpanContext
-          && (Utils.equals(referenceType, References.CHILD_OF)
-                      || Utils.equals(referenceType, References .FOLLOWS_FROM))) {
-        this.references.add(new Reference((SpanContext) referencedContext, referenceType));
-        this.baggage.putAll(((SpanContext)referencedContext).baggage());
+
+      if (!(referencedContext instanceof SpanContext)) {
+        return this;
+      }
+
+      // TODO UI currently does not support arbitrary references
```

Comment:
do we need this TODO? It's not actionable, and It's not just the UI, the Thrift model doesn't support other ref types: https://github.com/uber/jaeger-idl/blob/master/thrift/jaeger.thrift#L43